.changelog/README.md

```diff
@@ -0,0 +1,132 @@
+To install, run the following command:
+
+```
+go get github.com/hashicorp/go-changelog/cmd/changelog-build
+```
+
+## Developer Guide
+
+PRs with code changes should include a `.changelog/<PR Number>.txt` file highlighting the user visible change.
+
+The changelog entry should include the affected component (e.g. `driver/docker`, `api`), and a user-centric summary of the change. If an issue warrants more elaborate description, also update the [Upgrade Guide](../website/content/docs/upgrade/upgrade-specific.mdx).
+
+Use the PR number as the file name. Enterprise private changes should have an entry in the OSS repository referencing a public GitHub issues detailing the change and impact.
+
+Below are some examples of how to generate a CHANGELOG entry with your pull
+request.
+
+### Improvement
+
+~~~
+```release-note:improvement
+internal/server: Add new option for configs
+```
+~~~
+
+### Feature
+
+Significant major release features. Typically included in release blog announcement and keynotes.
+
+~~~
+```release-note:feature
+**Consul namespace support (Enterprise)**: Run Nomad-defined services in their HashiCorp Consul namespaces more easily using Nomad Enterpris
+```
+~~~
+
+### Improvements
+
+Improvements are incremental and quality of life enhancement that don't rise to the FEATURES level:
+
+```release-note:improvement
+cli: Added `-monitor` flag to `deployment status` command and automatically monitor deployments from `job run` command.
+```
+
+### Bug
+
+~~~
+```release-note:bug
+client/fingerprint/java: Fixed a bug where java fingerprinter would not detect some Java distributions
+```
+~~~
+
+### Multiple Entries
+
+If a PR has multiple entries,
+
+~~~
+```release-note:bug
+driver/docker: Fix broken code
+```
+```release-note:bug
+client: Fix broken code
+```
+```release-note:bug
+cli: Fix broken code
+```
+~~~
+
+### Long Description with Markdown
+
+~~~
+```release-note:feature
+main: Lorem ipsum dolor `sit amet`, _consectetur_ adipiscing elit, **sed** do
+eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
+cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+```
+~~~
+
+
+## How to generate CHANGELOG entries for release
+
+Below is an example for running `go-changelog` to generate a collection of
+entries. It will generate output that can be inserted into CHANGELOG.md manually.
+
+To generate latest release changelog entry:
+
+```
+make changelog
+```
+
+You can also call `changelog-build` binary directly. For more information as to what each flag does, make sure to run `changelog-build -help`.
+
+```
+changelog-build -last-release v1.1.2 -this-release HEAD \
+                -entries-dir . -changelog-template changelog.tmpl -note-template note.tmpl
+```
```

Comment:
This needs more constraints on when and where you're supposed to run this. As far as I can tell `HEAD` should be the release branch for the release, not `main`. Can we describe how this works for backports too?

Reply:
Yes - though, I'd say the detailed commands should belong in the nomad-releases scripts rather than the documentation here.

But even in backport, running `make changelog` on the backport branch should generate the backport. I've meant this line to highlight the full command options, not to guide how to use it. Maybe I can remove it from here if that causes confusion.

Reply:
> Yes - though, I'd say the detailed commands should belong in the nomad-releases scripts rather than the documentation here.

That totally makes sense, you're right.